common/src/main/java/com/wynntils/handlers/chat/type/RecipientType.java

```diff
@@ -4,30 +4,27 @@
  */
 package com.wynntils.handlers.chat.type;
 
-import com.wynntils.core.text.CodedString;
+import com.wynntils.core.text.StyledText;
 import java.util.regex.Pattern;
 
 public enum RecipientType {
     INFO(null, null, "Info"),
     CLIENTSIDE(null, null, "Clientside"),
     // https://regexr.com/7b14s
-    NPC(
-            "^(?:§r)?§7\\[\\d+\\/\\d+\\](?:§r§.)? ?§r§[25] ?.+: ?§r§..*$",
-            "^(?:§r)?§8\\[\\d+\\/\\d+\\] .+: ?§r§..*$",
-            "NPC"),
+    NPC("^§7\\[\\d+\\/\\d+\\](?:§.)? ?§[25] ?.+: ?§..*$", "^§8\\[\\d+\\/\\d+\\] .+: ?§..*$", "NPC"),
     GLOBAL(
-            "^§8\\[(Lv\\. )?\\d+\\*?/\\d+/..(/[^]]+)?\\]§r§7 \\[[A-Z0-9]+\\]§r.*$",
-            "^(§r§8)?\\[(Lv\\. )?\\d+\\*?/\\d+/..(/[^]]+)?\\] \\[[A-Z0-9]+\\](§r§7)?( \\[(§k\\|)?§r§.[A-Z+]+§r§.(§k\\|§r§7)?\\])?(§r§7)? (§r§8)?.*$",
+            "^§8\\[(Lv\\. )?\\d+\\*?/\\d+/..(/[^]]+)?\\]§7 \\[[A-Z0-9]+\\].*$",
+            "^(§8)?\\[(Lv\\. )?\\d+\\*?/\\d+/..(/[^]]+)?\\] \\[[A-Z0-9]+\\](§7)?( \\[(§k\\|)?§r§.[A-Z+]+§.(§k\\|§r§7)?\\])?(§7)? (§8)?.*$",
             "Global"),
     LOCAL(
-            "^§.\\[(Lv. )?\\d+\\*?/\\d+/..(/[^]]+)?\\]§r.*$",
-            "^(§r§8)?\\[(Lv. )?\\d+\\*?/\\d+/..(/[^]]+)?\\]( \\[(§k\\|)?§r§.[A-Z+]+§r§.(§k\\|§r§7)?\\])?(§r§7)? (§r§8)?.*$",
+            "^§.\\[(Lv. )?\\d+\\*?/\\d+/..(/[^]]+)?\\].*$",
+            "^(§8)?\\[(Lv. )?\\d+\\*?/\\d+/..(/[^]]+)?\\]( \\[(§k\\|)?§r§.[A-Z+]+.(§k\\|§r§7)?\\])?(§7)? (§8)?.*$",
             "Local"),
-    GUILD("^(§r)?§3\\[(§b★{0,5}§3)?.*§3]§. .*$", "^(§r§8)?\\[(§r§7★{0,5}§r§8)?.*]§r§7 .*$", "Guild"),
-    PARTY("^§7\\[§r§e[^➤]*§r§7\\] §r§f.*$", "^(§r§8)?\\[§r§7[^➤]*§r§8\\] §r§7[^§]*$", "Party"),
-    PRIVATE("^§7\\[.* ➤ .*\\] §r§f.*$", "^(§r§8)?\\[.* ➤ .*\\] §r§7.*$", "Private"),
-    SHOUT("^§3.* \\[[A-Z0-9]+\\] shouts: §r§b.*$", "^(§r§8)?.* \\[[A-Z0-9]+\\] shouts: §r§7.*$", "Shout"),
-    PETS("^§2(.*): §r§a(.*)$", "^§8(.*): §r§7(.*)$", "Pets");
```

Comment:
Did you just remove all resets or did you manually check these patterns (there are cases where the resets need to stay, you only need to remove when if they are followed by a **color** (not bold/italic/etc) code)?

Reply:
i did check. there are multiple §k and i didn't remove the following §r, but I've just noticed that some of the §k are optional and I didn't consider that